lib/helpers/tasks.rb

```diff
@@ -55,50 +55,48 @@ def run_command(command)
   end
 
   def generate_types(environment)
-    begin
-      command = "#{COMMAND_PREFIX} /opt/puppetlabs/puppet/bin generate types --environment #{environment}"
-
-      message = run_command(command)
-      LOGGER.info("message: #{message} environment: #{environment}")
-      status_message = {status: :success, message: message.to_s, environment: environment, status_code: 200 }
-      notify_slack(status_message) if slack?
-    rescue => e
-      LOGGER.error("message: #{e.message} trace: #{e.backtrace}")
-      status_message = {:status => :fail, :message => e.message, :trace => e.backtrace, :environment => environment, :status_code => 500}
-      notify_slack(status_message) if slack?
-    end
+    command = "#{COMMAND_PREFIX} /opt/puppetlabs/puppet/bin generate types --environment #{environment}"
+
+    message = run_command(command)
+    LOGGER.info("message: #{message} environment: #{environment}")
+    status_message = { status: :success, message: message.to_s, environment: environment, status_code: 200 }
+    notify_slack(status_message) if slack?
+  rescue StandardError => e
+    LOGGER.error("message: #{e.message} trace: #{e.backtrace}")
+    status_message = { status: :fail, message: e.message, trace: e.backtrace, environment: environment, status_code: 500 }
+    notify_slack(status_message) if slack?
   end
 
   def notify_slack(status_message)
     return unless settings.slack_webhook
 
-    if settings.slack_channel
-      slack_channel = settings.slack_channel
-    else
-      slack_channel = '#default'
-    end
+    slack_channel = if settings.slack_channel
```

Comment:
Can't this be ```slack_channel = settings.slack_channel || '#default'```?

Reply:
Unfortunately, it cannot. At least not yet.

The way `Sinatra::ConfigFile` works is that it will fail with a no method error if the config key doesn't exist in the yml file. My current workaround (which would allow this to work) is to have an empty key in the yml config file so that it will return nil.

I have some work to do on the config file front.

Reply:
Actually, nevermind. The `NoMethodError` would be triggered either way if the setting is missing. So, I agree with @ekohl, let's simplify that.

Reply:
Fixed.

Reply:
In general I like having a config file which has all the defaults in it rather than the code. It's a nice reference for users and can simplify code like this.